internal/system/agent/services.go

```diff
@@ -81,54 +86,94 @@ func InvokeOperation(action string, services []string) bool {
 	return true
 }
 
-func getConfig(services []string, ctx context.Context) (ConfigRespMap, error) {
+func getConfig(services []string, ctx context.Context) (map[string]interface{}, error) {
 
-	c := ConfigRespMap{}
-	c.Configuration = map[string]interface{}{}
+	configuration := make(map[string]interface{})
 
 	// Loop through requested actions, along with respectively-supplied parameters.
 	for _, service := range services {
 
-		c.Configuration[service] = ""
+		configuration[service] = ""
 
 		if !isKnownServiceKey(service) {
 			logs.LoggingClient.Warn("unknown service found getting configuration", "service name", service)
 		}
 
 		responseJSON, err := clients[service].FetchConfiguration(ctx)
 		if err != nil {
-			c.Configuration[service] = fmt.Sprintf("%s get config error: %s", service, err.Error())
+			configuration[service] = fmt.Sprintf("%s get config error: %s", service, err.Error())
 			logs.LoggingClient.Error("error retrieving configuration", "service name", service, "error message", err.Error())
 		} else {
-			c.Configuration[service] = ProcessResponse(responseJSON)
+			configuration[service] = ProcessResponse(responseJSON)
 		}
 	}
-	return c, nil
+	return configuration, nil
 }
 
-func getMetrics(services []string, ctx context.Context) (MetricsRespMap, error) {
+func getMetrics(services []string, ctx context.Context) (map[string]interface{}, error) {
 
-	m := MetricsRespMap{}
-	m.Metrics = map[string]interface{}{}
+	metrics := make(map[string]interface{})
 
 	// Loop through requested actions, along with respectively-supplied parameters.
 	for _, service := range services {
 
-		m.Metrics[service] = ""
+		metrics[service] = ""
```

Comment:
why does this get set here? It is reset to something else on all control flows below